ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/ConnectionProxy.java

```diff
@@ -75,53 +92,106 @@ public void close() throws JMSException {
         }
     }
 
+    /**
+     * Delegate to {@link #xaConnection}
```

Comment:
This javadoc comment is describing how the method is implemented. I would just duplicate the javadoc for whatever xaConnection.getClientID() says. Actually just saying @see Connection#getClientID() should be good enough. But it is up to you if you think that the text adds something.

Similarly for the other delegation methods below.